src/Tests/Microsoft.Diagnostics.Monitoring.EgressExtensibilityApp/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.Tool.UnitTests;
+using Microsoft.Diagnostics.Tools.Monitor.Egress;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Text.Json;
+
+namespace Microsoft.Diagnostics.Monitoring.EgressExtensibilityApp
+{
+    internal sealed class Program
+    {
+        static int Main(string[] args)
+        {
+            RootCommand rootCommand = new RootCommand();
+
+            Command egressCmd = new Command("Egress");
+
+            egressCmd.SetHandler(() => Egress());
+
+            rootCommand.Add(egressCmd);
+
+            return rootCommand.Invoke(args);
+        }
+
+        private static int Egress()
+        {
+            EgressArtifactResult result = new();
+            try
+            {
+                string jsonConfig = Console.ReadLine();
```

Comment:
`Console.ReadLine` and `Console.WriteLine` use line endings that are platform dependent e.g. Windows is `\r\n` and Unix-like (includes Linux and MacOS) is typically `\n`. I'm not sure that we want the protocol to be platform dependent. It might be okay since this is stdin/stdout on the same machine, but if we for some reason decide to remote the protocol usage (e.g. over the network), then the two machines aren't necessarily the same platform.

Reply:
Not sure if you wanted me to take action on this point, but I did want to point out that this particular example is for testing, not the actual extension - we can discuss this further if it's a concern though.

Reply:
I suppose that's okay to push down the road, but I don't want this to proliferate to additional copies and others start playing with extensions assuming this is okay.